crates/bevy_tasks/src/task_pool.rs

```diff
@@ -95,56 +105,67 @@ impl TaskPool {
         TaskPoolBuilder::new().build()
     }
 
+    #[allow(unused_variables)]
     fn new_internal(
```

Comment:
I think these implementations are different enough that they don't benefit from being side-by-side. Instead I think it would be clearer if we moved each to its own module, then selected the module based on `cfg`:

```rust
// lib.rs

#[cfg(not(target_arch = "wasm32"))]
mod task_pool;

#[cfg(not(target_arch = "wasm32"))]
pub use task_pool::TaskPool;

#[cfg(target_arch = "wasm32")]
mod single_threaded_task_pool::TaskPool;

#[cfg(target_arch = "wasm32")]
pub use single_threaded_task_pool::TaskPool;
```

Reply:
I started it like so, but ended copy-pasting too much.

But I guess at the current point, the implementations diverged so much, that it makes sense to split to separate files.
I will make this change.

Reply:
It indeed looks much nicer. Thanks for the suggestion.